src/main/java/in/projecteka/consentmanager/common/cache/ICacheAdapter.java

```diff
@@ -0,0 +1,11 @@
+package in.projecteka.consentmanager.common.cache;
+
+public interface ICacheAdapter<K,V> {
+    V get(K key);
+
+    void put(K key, V value);
```

Comment:
Earlier it was fine to use sync cache update since it was in-memory cache,

Is it possible to define an abstraction in reactive way and covert the 
in-memory and redis cache to use reactive APIs

 https://github.com/lettuce-io/lettuce-core#reactive-api

Reply:
I can look into this!

Reply:
Updated with a reactive interface for caching and also a bit of refactoring to accommodate these changes